### PR TITLE
ui 190 deposit and withdrawal history

### PR DESCRIPTION
**Pull request by Prithvi101**
- **feat: added base table component**
- **feat: added deposit and order rows to be used in base table**
- **feat: added actiity center to use base table in tabs**
- **chore: restructured trade page to use flexbox instade of grid**
- **style: fixed minor styling related to trade page**
image
image

### CODE REVIEW

File: src/app/portfolio/page.tsx [hunk 1]
```diff
@@ -43,8 +43,8 @@ const PortfolioPage = () => {
     }, [getUserDepositBalance, isDepositModalOpen, isWithdrawModalOpen])
 
     const handleDeposit = () => {
-        console.log('Deposit action triggered');
-    };
+        console.log("Deposit action triggered")
+    }
```
Comment (gopherine) [inline src/app/portfolio/page.tsx:47]: is this only being used for logging right now ? or is awaiting backend integration what is the purpose of this function?

File: src/app/trade/page.tsx [hunk 2]
```diff
@@ -16,23 +16,34 @@ import { useLimitPrice } from "@/hooks/market-data/useLimitPrice"
...
-        <div className='grid grid-cols-12 grid-rows-12 w-full  gap-3 '>
-            <div className='col-span-6 3xl:col-span-7 4xl:col-span-7 row-span-8   '>
-                <TradeChartContainer />
-            </div>
-            {/* Orderbook and PayOff Graph Section */}
-            <div className=' col-span-3 3xl:col-span-2 row-span-8  bg-[#0F121A]  rounded-[8px]'>
-                <TradeInsights></TradeInsights>
-            </div>
+        <div className='flex h-auto w-full min-w-[100vw]  gap-4'>
+            {/* Trade View */}
```
Comment (gopherine) [inline src/app/trade/page.tsx:28]: I find these comments a bit pointless ? whats your opinion here ? i feel that we are anyways importing components and there names clearly signifies what we are doing for each block having line breaks should be enough for readability ?

File: src/component/Blocks/Global/AccountOverview/ActivityCenter.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        value: "depostiAndWithdrawls",
+        data: {
+            header: DepostiHeaders,
+            rows: DepostiDetails,
+        },
+    },
+]
+
+function ActivityCenter() {
+    const [selectedTab, setSelectedTab] = useState("positions")
```
Comment (gopherine) [inline src/component/Blocks/Global/AccountOverview/ActivityCenter.tsx:48]: NIT:

i feel use of enums here would be preferred here for values like positions,openOrders etc.
though this works but addition might maintain consistency

File: src/component/Ui/Tables/BaseTable.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { Table, TableBody, TableHead, TableHeader, TableRow } from "@/components/ui/table"
+import React from "react"
+
+export interface BaseTableProps {
+    tableHead: string[] // Array of table headers
+    children: React.ReactNode // Children for the TableBody
```
Comment (gopherine) [inline src/component/Ui/Tables/BaseTable.tsx]: comment seems to be redundant here ? variable names just already clearly defines it

File: src/component/Ui/Tables/DepositsRow.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+export interface DepositsRowProps {
+    rowIndex: number
+    row: DepostisRowDetails
+}
+
+const StatusColors = {
+    Completed: "#28a745", // Green
+    Pending: "#ff8c00", // Yellow-orange
+    Failed: "#dc3545", // Red
```
Comment (gopherine) [inline src/component/Ui/Tables/DepositsRow.tsx:12]: I would suggest that colors should be part of global constants- and should be ported from there for consistency and ease of access

**Review by gopherine — CHANGES_REQUESTED** [review]

File: src/app/trade/page.tsx (hunk 2 above)
Comment (Prithvi101) [reply src/app/trade/page.tsx:28]: Yes it's but this is page section and as we add more  and more components it get's difficult to navigate I have habbit ho collapsing divs that i am not working on so comments are only way i can track components  like in the below example
image

File: src/component/Blocks/Global/AccountOverview/ActivityCenter.tsx (hunk 3 above)
Comment (Prithvi101) [reply src/component/Blocks/Global/AccountOverview/ActivityCenter.tsx:48]: actually It's just dummy data that i am adding actual data will be from hook or api so didn't bother to do lot of effort on it

File: src/component/Ui/Tables/BaseTable.tsx (hunk 4 above)
Comment (Prithvi101) [reply src/component/Ui/Tables/BaseTable.tsx]: GPT xD will remove them

File: src/component/Ui/Tables/DepositsRow.tsx (hunk 5 above)
Comment (Prithvi101) [reply src/component/Ui/Tables/DepositsRow.tsx:12]: yes will do that when i m working on styling for now there is no source for colors and style so just hardcoded them

File: src/app/portfolio/page.tsx (hunk 1 above)
Comment (Prithvi101) [reply src/app/portfolio/page.tsx:47]: not sure haven't checked it it's here because of pritier styled it

Comment (Prithvi101) [reply src/app/portfolio/page.tsx:47]: i believe inetgration is pending

**Review by gopherine — APPROVED** [review]

**Pull request merged**